tests/ui/identity_op.rs

```diff
@@ -77,4 +77,20 @@ fn main() {
     (x + 1) % 3; // no error
     4 % 3; // no error
     4 % -3; // no error
+
+    // See #8724
+    let a = 0;
+    let b = true;
+    0 + if b { 1 } else { 2 };
```

Comment:
I would like to see a testcase like `let _c = 0 + if b { 1 } else { 2 };`
Maybe also a function like I wrote in #8724:
```rust
pub fn decide(a: bool, b: bool) -> u32 {
    0 + if a { 1 } else { 2 } + if b { 3 } else { 5 }
}
```